R/data_to_wide.R

```diff
@@ -258,6 +258,16 @@ data_to_wide <- function(data,
     attributes(out[[i]]) <- variable_attr[[i]]
   }
 
+  # convert back to date if original values were dates
+  values_are_dates <- all(vapply(data[, values_from, drop = FALSE], function(x) {
+    inherits(x, "Date")
+  }, FUN.VALUE = logical(1)))
```

Comment:
Nit: Extract out the anonymous function to a helper `.is_date()`, and use that instead.

Ideally, `apply()` calls should be one-lines to improve readability.